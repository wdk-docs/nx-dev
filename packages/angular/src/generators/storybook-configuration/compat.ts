import { convertNxGenerator } from '@nrwl/devkit';
import { storybookConfigurationGenerator } from './storybook-configuration';

export default convertNxGenerator(storybookConfigurationGenerator);
