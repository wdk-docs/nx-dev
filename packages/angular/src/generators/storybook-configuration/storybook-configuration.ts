import type { GeneratorCallback, Tree } from '@nrwl/devkit';
import { assertCompatibleStorybookVersion } from './lib/assert-compatible-storybook-version';
import { generateStories } from './lib/generate-stories';
import { generateStorybookConfiguration } from './lib/generate-storybook-configuration';
import { validateOptions } from './lib/validate-options';
import type { StorybookConfigurationOptions } from './schema';

export async function storybookConfigurationGenerator(
  tree: Tree,
  options: StorybookConfigurationOptions
): Promise<GeneratorCallback> {
  assertCompatibleStorybookVersion();
  validateOptions(options);

  const storybookGeneratorInstallTask = await generateStorybookConfiguration(
    tree,
    options
  );

  if (options.generateStories) {
    generateStories(tree, options);
  }

  return () => {
    storybookGeneratorInstallTask();
  };
}

export default storybookConfigurationGenerator;
