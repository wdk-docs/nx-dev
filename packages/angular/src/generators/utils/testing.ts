import type { Tree } from '@nrwl/devkit';
import { wrapAngularDevkitSchematic } from '@nrwl/devkit/ngcli-adapter';
import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing';
import { Linter } from '@nrwl/linter';
import { UnitTestRunner } from '../../utils/test-runners';
import libraryGenerator from '../library/library';

export async function createStorybookTestWorkspaceForLib(
  libName: string
): Promise<Tree> {
  let tree = createTreeWithEmptyWorkspace();

  const moduleGenerator = wrapAngularDevkitSchematic(
    '@schematics/angular',
    'module'
  );
  const componentGenerator = wrapAngularDevkitSchematic(
    '@schematics/angular',
    'component'
  );

  await libraryGenerator(tree, {
    name: libName,
    buildable: false,
    linter: Linter.EsLint,
    publishable: false,
    simpleModuleName: false,
    skipFormat: false,
    unitTestRunner: UnitTestRunner.Jest,
  });

  await componentGenerator(tree, {
    name: 'test-button',
    project: libName,
  });

  tree.write(
    `libs/${libName}/src/lib/test-button/test-button.component.ts`,
    `import { Component, Input } from '@angular/core';

export type ButtonStyle = 'default' | 'primary' | 'accent';

@Component({
  selector: 'proj-test-button',
  templateUrl: './test-button.component.html',
  styleUrls: ['./test-button.component.css']
})
export class TestButtonComponent {
  @Input('buttonType') type = 'button';
  @Input() style: ButtonStyle = 'default';
  @Input() age?: number;
  @Input() isOn = false;
}`
  );

  tree.write(
    `libs/${libName}/src/lib/test-button/test-button.component.html`,
    `<button [attr.type]="type" [ngClass]="style"></button>`
  );

  const modulePath = `libs/${libName}/src/lib/${libName}.module.ts`;
  tree.write(
    modulePath,
    `import * as ButtonExports from './test-button/test-button.component';
    ${tree.read(modulePath)}`
  );

  // create a module with component that gets exported in a barrel file
  await moduleGenerator(tree, {
    name: 'barrel',
    project: libName,
  });

  await componentGenerator(tree, {
    name: 'barrel-button',
    project: libName,
    path: `libs/${libName}/src/lib/barrel`,
    module: 'barrel',
  });

  tree.write(
    `libs/${libName}/src/lib/barrel/barrel-button/index.ts`,
    `export * from './barrel-button.component';`
  );

  tree.write(
    `libs/${libName}/src/lib/barrel/barrel.module.ts`,
    `import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BarrelButtonComponent } from './barrel-button';

@NgModule({
  imports: [CommonModule],
  declarations: [BarrelButtonComponent],
})
export class BarrelModule {}`
  );

  // create a module with components that get Angular exported and declared by variable
  await moduleGenerator(tree, {
    name: 'variable-declare',
    project: libName,
  });

  await componentGenerator(tree, {
    name: 'variable-declare-button',
    project: libName,
    path: `libs/${libName}/src/lib/variable-declare`,
    module: 'variable-declare',
  });

  await componentGenerator(tree, {
    name: 'variable-declare-view',
    project: libName,
    path: `libs/${libName}/src/lib/variable-declare`,
    module: 'variable-declare',
  });

  tree.write(
    `libs/${libName}/src/lib/variable-declare/variable-declare.module.ts`,
    `import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { VariableDeclareButtonComponent } from './variable-declare-button/variable-declare-button.component';
import { VariableDeclareViewComponent } from './variable-declare-view/variable-declare-view.component';

const COMPONENTS = [
  VariableDeclareButtonComponent,
  VariableDeclareViewComponent
]

@NgModule({
  imports: [CommonModule],
  declarations: COMPONENTS,
  exports: COMPONENTS
})
export class VariableDeclareModule {}`
  );

  // create a module with components that get Angular exported and declared by variable
  await moduleGenerator(tree, {
    name: 'variable-spread-declare',
    project: libName,
  });

  await componentGenerator(tree, {
    name: 'variable-spread-declare-button',
    project: libName,
    path: `libs/${libName}/src/lib/variable-spread-declare`,
    module: 'variable-spread-declare',
  });

  await componentGenerator(tree, {
    name: 'variable-spread-declare-view',
    project: libName,
    path: `libs/${libName}/src/lib/variable-spread-declare`,
    module: 'variable-spread-declare',
  });

  await componentGenerator(tree, {
    name: 'variable-spread-declare-anotherview',
    project: libName,
    path: `libs/${libName}/src/lib/variable-spread-declare`,
    module: 'variable-spread-declare',
  });

  tree.write(
    `libs/${libName}/src/lib/variable-spread-declare/variable-spread-declare.module.ts`,
    `import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { VariableSpreadDeclareButtonComponent } from './variable-spread-declare-button/variable-spread-declare-button.component';
import { VariableSpreadDeclareViewComponent } from './variable-spread-declare-view/variable-spread-declare-view.component';
import { VariableSpreadDeclareAnotherviewComponent } from './variable-spread-declare-anotherview/variable-spread-declare-anotherview.component';

const COMPONENTS = [ 
  VariableSpreadDeclareButtonComponent, 
  VariableSpreadDeclareViewComponent 
]

@NgModule({
  imports: [CommonModule],
  declarations: [...COMPONENTS, VariableSpreadDeclareAnotherviewComponent],
})
export class VariableSpreadDeclareModule {}`
  );

  // create a module where declared components are pulled from a static member of the module
  await moduleGenerator(tree, {
    name: 'static-member-declarations',
    project: libName,
  });

  await componentGenerator(tree, {
    name: 'cmp1',
    project: libName,
    path: `libs/${libName}/src/lib/static-member-declarations`,
    module: 'static-member-declarations',
  });

  await componentGenerator(tree, {
    name: 'cmp2',
    project: libName,
    path: `libs/${libName}/src/lib/static-member-declarations`,
    module: 'static-member-declarations',
  });

  tree.write(
    `libs/${libName}/src/lib/static-member-declarations/static-member-declarations.module.ts`,
    `import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { Cmp1Component } from './cmp1/cmp1.component';
import { Cmp2Component } from './cmp2/cmp2.component';

@NgModule({
  imports: [CommonModule],
  declarations: StaticMemberDeclarationsModule.COMPONENTS,
  exports: StaticMemberDeclarationsModule.COMPONENTS
})
export class StaticMemberDeclarationsModule {
  static readonly COMPONENTS = [Cmp1Component, Cmp2Component];
}`
  );

  // create another button in a nested subpath
  await moduleGenerator(tree, {
    name: 'nested',
    project: libName,
    path: `libs/${libName}/src/lib`,
  });

  await componentGenerator(tree, {
    name: 'nested-button',
    project: libName,
    module: 'nested',
    path: `libs/${libName}/src/lib/nested`,
  });

  await componentGenerator(tree, {
    name: 'test-other',
    project: libName,
  });

  return tree;
}
