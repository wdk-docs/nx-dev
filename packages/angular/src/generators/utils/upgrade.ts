import type { GeneratorCallback, Tree } from '@nrwl/devkit';
import { installPackagesTask, updateJson } from '@nrwl/devkit';
import { angularJsVersion } from '../../utils/versions';

export function addUpgradeToPackageJson(tree: Tree): GeneratorCallback {
  updateJson(tree, './package.json', (json) => {
    json.dependencies = {
      '@angular/upgrade': json.dependencies['@angular/core'],
      angular: angularJsVersion,
      ...(json.dependencies ?? {}),
    };
    json.dependencies = sortObjectByKeys(json.dependencies);

    return json;
  });

  return () => {
    installPackagesTask(tree);
  };
}

function sortObjectByKeys(obj: unknown): unknown {
  return Object.keys(obj)
    .sort()
    .reduce((result, key) => {
      return {
        ...result,
        [key]: obj[key],
      };
    }, {});
}
