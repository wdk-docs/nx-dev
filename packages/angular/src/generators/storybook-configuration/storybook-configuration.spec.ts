import type { Tree } from '@nrwl/devkit';
import { joinPathFragments } from '@nrwl/devkit';
import { overrideCollectionResolutionForTesting } from '@nrwl/devkit/ngcli-adapter';
import * as fileUtils from '@nrwl/workspace/src/core/file-utils';
import { Linter } from 'packages/linter/src/generators/utils/linter';
import { createStorybookTestWorkspaceForLib } from '../utils/testing';
import type { StorybookConfigurationOptions } from './schema';
import { storybookConfigurationGenerator } from './storybook-configuration';

function listFiles(tree: Tree): string[] {
  const files = new Set<string>();
  tree.listChanges().forEach((change) => {
    if (change.type !== 'DELETE') {
      files.add(change.path);
    }
  });

  return Array.from(files).sort((a, b) => a.localeCompare(b));
}

describe('StorybookConfiguration generator', () => {
  let tree: Tree;
  const libName = 'test-ui-lib';

  beforeEach(async () => {
    tree = await createStorybookTestWorkspaceForLib(libName);

    overrideCollectionResolutionForTesting({
      '@nrwl/storybook': joinPathFragments(
        __dirname,
        '../../../../storybook/generators.json'
      ),
    });
    jest.resetModules();
    jest.doMock('@storybook/angular/package.json', () => ({
      version: '6.4.0-rc.1',
    }));
    jest.spyOn(fileUtils, 'readPackageJson').mockReturnValue({
      devDependencies: {
        '@storybook/addon-essentials': '~6.2.9',
        '@storybook/react': '~6.2.9',
      },
    });
  });

  it('should throw when the @storybook/angular version is lower than 6.4.0-rc.1', async () => {
    jest.doMock('@storybook/angular/package.json', () => ({
      version: '5.1.0',
    }));

    await expect(
      storybookConfigurationGenerator(tree, <StorybookConfigurationOptions>{
        name: libName,
      })
    ).rejects.toThrowErrorMatchingInlineSnapshot(
      '"Incompatible Storybook Version: Please use a version of @storybook/angular higher than 6.4.0-rc.1"'
    );
  });

  it('should throw when generateCypressSpecs is true and generateStories is false', async () => {
    await expect(
      storybookConfigurationGenerator(tree, <StorybookConfigurationOptions>{
        name: libName,
        generateCypressSpecs: true,
        generateStories: false,
      })
    ).rejects.toThrow(
      'Cannot set generateCypressSpecs to true when generateStories is set to false.'
    );
  });

  it('should only configure storybook', async () => {
    await storybookConfigurationGenerator(tree, <StorybookConfigurationOptions>{
      name: libName,
      configureCypress: false,
      generateCypressSpecs: false,
      generateStories: false,
    });

    expect(tree.exists('libs/test-ui-lib/.storybook/main.js')).toBeTruthy();
    expect(
      tree.exists('libs/test-ui-lib/.storybook/tsconfig.json')
    ).toBeTruthy();
    expect(tree.exists('apps/test-ui-lib-e2e/cypress.json')).toBeFalsy();
    expect(
      tree.exists(
        'libs/test-ui-lib/src/lib/test-button/test-button.component.stories.ts'
      )
    ).toBeFalsy();
    expect(
      tree.exists(
        'libs/test-ui-lib/src/lib/test-other/test-other.component.stories.ts'
      )
    ).toBeFalsy();
    expect(
      tree.exists(
        'apps/test-ui-lib-e2e/src/integration/test-button/test-button.component.spec.ts'
      )
    ).toBeFalsy();
    expect(
      tree.exists(
        'apps/test-ui-lib-e2e/src/integration/test-other/test-other.component.spec.ts'
      )
    ).toBeFalsy();
  });

  it('should configure storybook to use webpack 5', async () => {
    await storybookConfigurationGenerator(tree, {
      name: libName,
      configureCypress: false,
      generateCypressSpecs: false,
      generateStories: false,
      linter: Linter.None,
    });

    expect(
      tree.read('libs/test-ui-lib/.storybook/main.js').toString()
    ).toMatchSnapshot();
  });

  it('should configure everything at once', async () => {
    await storybookConfigurationGenerator(tree, <StorybookConfigurationOptions>{
      name: libName,
      configureCypress: true,
      generateCypressSpecs: true,
      generateStories: true,
    });

    expect(tree.exists('libs/test-ui-lib/.storybook/main.js')).toBeTruthy();
    expect(
      tree.exists('libs/test-ui-lib/.storybook/tsconfig.json')
    ).toBeTruthy();
    expect(tree.exists('apps/test-ui-lib-e2e/cypress.json')).toBeTruthy();
    expect(
      tree.exists(
        'libs/test-ui-lib/src/lib/test-button/test-button.component.stories.ts'
      )
    ).toBeTruthy();
    expect(
      tree.exists(
        'libs/test-ui-lib/src/lib/test-other/test-other.component.stories.ts'
      )
    ).toBeTruthy();
    expect(
      tree.exists(
        'apps/test-ui-lib-e2e/src/integration/test-button/test-button.component.spec.ts'
      )
    ).toBeTruthy();
    expect(
      tree.exists(
        'apps/test-ui-lib-e2e/src/integration/test-other/test-other.component.spec.ts'
      )
    ).toBeTruthy();
  });

  it('should generate the right files', async () => {
    await storybookConfigurationGenerator(tree, <StorybookConfigurationOptions>{
      name: libName,
      configureCypress: true,
      generateCypressSpecs: true,
      generateStories: true,
    });

    expect(listFiles(tree)).toMatchSnapshot();
  });

  it('should generate in the correct folder', async () => {
    await storybookConfigurationGenerator(tree, <StorybookConfigurationOptions>{
      name: libName,
      configureCypress: true,
      generateCypressSpecs: true,
      generateStories: true,
      cypressDirectory: 'one/two',
    });

    expect(listFiles(tree)).toMatchSnapshot();
  });
});
