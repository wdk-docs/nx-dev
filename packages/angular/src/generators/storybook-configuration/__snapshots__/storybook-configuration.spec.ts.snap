// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StorybookConfiguration generator should configure storybook to use webpack 5 1`] = `
"const rootMain = require('../../../.storybook/main');


module.exports = {
  ...rootMain,
  
  core: { ...rootMain.core, builder: 'webpack5' },
  
  stories: [
    ...rootMain.stories,
    '../src/lib/**/*.stories.mdx',
    '../src/lib/**/*.stories.@(js|jsx|ts|tsx)'
  ],
  addons: [...rootMain.addons ],
  webpackFinal: async (config, { configType }) => {
    // apply any global webpack configs that might have been specified in .storybook/main.js
    if (rootMain.webpackFinal) {
      config = await rootMain.webpackFinal(config, { configType });
    }
    
    

    // add your own webpack tweaks if needed

    return config;
  },
};
"
`;

exports[`StorybookConfiguration generator should generate in the correct folder 1`] = `
Array [
  ".eslintrc.json",
  ".prettierrc",
  ".storybook/main.js",
  ".storybook/tsconfig.json",
  "apps/one/two/test-ui-lib-e2e/.eslintrc.json",
  "apps/one/two/test-ui-lib-e2e/cypress.json",
  "apps/one/two/test-ui-lib-e2e/src/fixtures/example.json",
  "apps/one/two/test-ui-lib-e2e/src/integration/barrel-button/barrel-button.component.spec.ts",
  "apps/one/two/test-ui-lib-e2e/src/integration/cmp1/cmp1.component.spec.ts",
  "apps/one/two/test-ui-lib-e2e/src/integration/cmp2/cmp2.component.spec.ts",
  "apps/one/two/test-ui-lib-e2e/src/integration/nested-button/nested-button.component.spec.ts",
  "apps/one/two/test-ui-lib-e2e/src/integration/test-button/test-button.component.spec.ts",
  "apps/one/two/test-ui-lib-e2e/src/integration/test-other/test-other.component.spec.ts",
  "apps/one/two/test-ui-lib-e2e/src/integration/variable-declare-button/variable-declare-button.component.spec.ts",
  "apps/one/two/test-ui-lib-e2e/src/integration/variable-declare-view/variable-declare-view.component.spec.ts",
  "apps/one/two/test-ui-lib-e2e/src/integration/variable-spread-declare-anotherview/variable-spread-declare-anotherview.component.spec.ts",
  "apps/one/two/test-ui-lib-e2e/src/integration/variable-spread-declare-button/variable-spread-declare-button.component.spec.ts",
  "apps/one/two/test-ui-lib-e2e/src/integration/variable-spread-declare-view/variable-spread-declare-view.component.spec.ts",
  "apps/one/two/test-ui-lib-e2e/src/support/commands.ts",
  "apps/one/two/test-ui-lib-e2e/src/support/index.ts",
  "apps/one/two/test-ui-lib-e2e/tsconfig.json",
  "jest.config.js",
  "jest.preset.js",
  "libs/test-ui-lib/.browserslistrc",
  "libs/test-ui-lib/.eslintrc.json",
  "libs/test-ui-lib/.storybook/main.js",
  "libs/test-ui-lib/.storybook/preview.js",
  "libs/test-ui-lib/.storybook/tsconfig.json",
  "libs/test-ui-lib/jest.config.js",
  "libs/test-ui-lib/README.md",
  "libs/test-ui-lib/src/index.ts",
  "libs/test-ui-lib/src/lib/barrel/barrel-button/barrel-button.component.css",
  "libs/test-ui-lib/src/lib/barrel/barrel-button/barrel-button.component.html",
  "libs/test-ui-lib/src/lib/barrel/barrel-button/barrel-button.component.spec.ts",
  "libs/test-ui-lib/src/lib/barrel/barrel-button/barrel-button.component.stories.ts",
  "libs/test-ui-lib/src/lib/barrel/barrel-button/barrel-button.component.ts",
  "libs/test-ui-lib/src/lib/barrel/barrel-button/index.ts",
  "libs/test-ui-lib/src/lib/barrel/barrel.module.ts",
  "libs/test-ui-lib/src/lib/nested/nested-button/nested-button.component.css",
  "libs/test-ui-lib/src/lib/nested/nested-button/nested-button.component.html",
  "libs/test-ui-lib/src/lib/nested/nested-button/nested-button.component.spec.ts",
  "libs/test-ui-lib/src/lib/nested/nested-button/nested-button.component.stories.ts",
  "libs/test-ui-lib/src/lib/nested/nested-button/nested-button.component.ts",
  "libs/test-ui-lib/src/lib/nested/nested.module.ts",
  "libs/test-ui-lib/src/lib/static-member-declarations/cmp1/cmp1.component.css",
  "libs/test-ui-lib/src/lib/static-member-declarations/cmp1/cmp1.component.html",
  "libs/test-ui-lib/src/lib/static-member-declarations/cmp1/cmp1.component.spec.ts",
  "libs/test-ui-lib/src/lib/static-member-declarations/cmp1/cmp1.component.stories.ts",
  "libs/test-ui-lib/src/lib/static-member-declarations/cmp1/cmp1.component.ts",
  "libs/test-ui-lib/src/lib/static-member-declarations/cmp2/cmp2.component.css",
  "libs/test-ui-lib/src/lib/static-member-declarations/cmp2/cmp2.component.html",
  "libs/test-ui-lib/src/lib/static-member-declarations/cmp2/cmp2.component.spec.ts",
  "libs/test-ui-lib/src/lib/static-member-declarations/cmp2/cmp2.component.stories.ts",
  "libs/test-ui-lib/src/lib/static-member-declarations/cmp2/cmp2.component.ts",
  "libs/test-ui-lib/src/lib/static-member-declarations/static-member-declarations.module.ts",
  "libs/test-ui-lib/src/lib/test-button/test-button.component.css",
  "libs/test-ui-lib/src/lib/test-button/test-button.component.html",
  "libs/test-ui-lib/src/lib/test-button/test-button.component.spec.ts",
  "libs/test-ui-lib/src/lib/test-button/test-button.component.stories.ts",
  "libs/test-ui-lib/src/lib/test-button/test-button.component.ts",
  "libs/test-ui-lib/src/lib/test-other/test-other.component.css",
  "libs/test-ui-lib/src/lib/test-other/test-other.component.html",
  "libs/test-ui-lib/src/lib/test-other/test-other.component.spec.ts",
  "libs/test-ui-lib/src/lib/test-other/test-other.component.stories.ts",
  "libs/test-ui-lib/src/lib/test-other/test-other.component.ts",
  "libs/test-ui-lib/src/lib/test-ui-lib.module.ts",
  "libs/test-ui-lib/src/lib/variable-declare/variable-declare-button/variable-declare-button.component.css",
  "libs/test-ui-lib/src/lib/variable-declare/variable-declare-button/variable-declare-button.component.html",
  "libs/test-ui-lib/src/lib/variable-declare/variable-declare-button/variable-declare-button.component.spec.ts",
  "libs/test-ui-lib/src/lib/variable-declare/variable-declare-button/variable-declare-button.component.stories.ts",
  "libs/test-ui-lib/src/lib/variable-declare/variable-declare-button/variable-declare-button.component.ts",
  "libs/test-ui-lib/src/lib/variable-declare/variable-declare-view/variable-declare-view.component.css",
  "libs/test-ui-lib/src/lib/variable-declare/variable-declare-view/variable-declare-view.component.html",
  "libs/test-ui-lib/src/lib/variable-declare/variable-declare-view/variable-declare-view.component.spec.ts",
  "libs/test-ui-lib/src/lib/variable-declare/variable-declare-view/variable-declare-view.component.stories.ts",
  "libs/test-ui-lib/src/lib/variable-declare/variable-declare-view/variable-declare-view.component.ts",
  "libs/test-ui-lib/src/lib/variable-declare/variable-declare.module.ts",
  "libs/test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-anotherview/variable-spread-declare-anotherview.component.css",
  "libs/test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-anotherview/variable-spread-declare-anotherview.component.html",
  "libs/test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-anotherview/variable-spread-declare-anotherview.component.spec.ts",
  "libs/test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-anotherview/variable-spread-declare-anotherview.component.stories.ts",
  "libs/test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-anotherview/variable-spread-declare-anotherview.component.ts",
  "libs/test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-button/variable-spread-declare-button.component.css",
  "libs/test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-button/variable-spread-declare-button.component.html",
  "libs/test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-button/variable-spread-declare-button.component.spec.ts",
  "libs/test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-button/variable-spread-declare-button.component.stories.ts",
  "libs/test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-button/variable-spread-declare-button.component.ts",
  "libs/test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-view/variable-spread-declare-view.component.css",
  "libs/test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-view/variable-spread-declare-view.component.html",
  "libs/test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-view/variable-spread-declare-view.component.spec.ts",
  "libs/test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-view/variable-spread-declare-view.component.stories.ts",
  "libs/test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-view/variable-spread-declare-view.component.ts",
  "libs/test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare.module.ts",
  "libs/test-ui-lib/src/test-setup.ts",
  "libs/test-ui-lib/tsconfig.json",
  "libs/test-ui-lib/tsconfig.lib.json",
  "libs/test-ui-lib/tsconfig.spec.json",
  "nx.json",
  "package.json",
  "tsconfig.base.json",
  "workspace.json",
]
`;

exports[`StorybookConfiguration generator should generate the right files 1`] = `
Array [
  ".eslintrc.json",
  ".prettierrc",
  ".storybook/main.js",
  ".storybook/tsconfig.json",
  "apps/test-ui-lib-e2e/.eslintrc.json",
  "apps/test-ui-lib-e2e/cypress.json",
  "apps/test-ui-lib-e2e/src/fixtures/example.json",
  "apps/test-ui-lib-e2e/src/integration/barrel-button/barrel-button.component.spec.ts",
  "apps/test-ui-lib-e2e/src/integration/cmp1/cmp1.component.spec.ts",
  "apps/test-ui-lib-e2e/src/integration/cmp2/cmp2.component.spec.ts",
  "apps/test-ui-lib-e2e/src/integration/nested-button/nested-button.component.spec.ts",
  "apps/test-ui-lib-e2e/src/integration/test-button/test-button.component.spec.ts",
  "apps/test-ui-lib-e2e/src/integration/test-other/test-other.component.spec.ts",
  "apps/test-ui-lib-e2e/src/integration/variable-declare-button/variable-declare-button.component.spec.ts",
  "apps/test-ui-lib-e2e/src/integration/variable-declare-view/variable-declare-view.component.spec.ts",
  "apps/test-ui-lib-e2e/src/integration/variable-spread-declare-anotherview/variable-spread-declare-anotherview.component.spec.ts",
  "apps/test-ui-lib-e2e/src/integration/variable-spread-declare-button/variable-spread-declare-button.component.spec.ts",
  "apps/test-ui-lib-e2e/src/integration/variable-spread-declare-view/variable-spread-declare-view.component.spec.ts",
  "apps/test-ui-lib-e2e/src/support/commands.ts",
  "apps/test-ui-lib-e2e/src/support/index.ts",
  "apps/test-ui-lib-e2e/tsconfig.json",
  "jest.config.js",
  "jest.preset.js",
  "libs/test-ui-lib/.browserslistrc",
  "libs/test-ui-lib/.eslintrc.json",
  "libs/test-ui-lib/.storybook/main.js",
  "libs/test-ui-lib/.storybook/preview.js",
  "libs/test-ui-lib/.storybook/tsconfig.json",
  "libs/test-ui-lib/jest.config.js",
  "libs/test-ui-lib/README.md",
  "libs/test-ui-lib/src/index.ts",
  "libs/test-ui-lib/src/lib/barrel/barrel-button/barrel-button.component.css",
  "libs/test-ui-lib/src/lib/barrel/barrel-button/barrel-button.component.html",
  "libs/test-ui-lib/src/lib/barrel/barrel-button/barrel-button.component.spec.ts",
  "libs/test-ui-lib/src/lib/barrel/barrel-button/barrel-button.component.stories.ts",
  "libs/test-ui-lib/src/lib/barrel/barrel-button/barrel-button.component.ts",
  "libs/test-ui-lib/src/lib/barrel/barrel-button/index.ts",
  "libs/test-ui-lib/src/lib/barrel/barrel.module.ts",
  "libs/test-ui-lib/src/lib/nested/nested-button/nested-button.component.css",
  "libs/test-ui-lib/src/lib/nested/nested-button/nested-button.component.html",
  "libs/test-ui-lib/src/lib/nested/nested-button/nested-button.component.spec.ts",
  "libs/test-ui-lib/src/lib/nested/nested-button/nested-button.component.stories.ts",
  "libs/test-ui-lib/src/lib/nested/nested-button/nested-button.component.ts",
  "libs/test-ui-lib/src/lib/nested/nested.module.ts",
  "libs/test-ui-lib/src/lib/static-member-declarations/cmp1/cmp1.component.css",
  "libs/test-ui-lib/src/lib/static-member-declarations/cmp1/cmp1.component.html",
  "libs/test-ui-lib/src/lib/static-member-declarations/cmp1/cmp1.component.spec.ts",
  "libs/test-ui-lib/src/lib/static-member-declarations/cmp1/cmp1.component.stories.ts",
  "libs/test-ui-lib/src/lib/static-member-declarations/cmp1/cmp1.component.ts",
  "libs/test-ui-lib/src/lib/static-member-declarations/cmp2/cmp2.component.css",
  "libs/test-ui-lib/src/lib/static-member-declarations/cmp2/cmp2.component.html",
  "libs/test-ui-lib/src/lib/static-member-declarations/cmp2/cmp2.component.spec.ts",
  "libs/test-ui-lib/src/lib/static-member-declarations/cmp2/cmp2.component.stories.ts",
  "libs/test-ui-lib/src/lib/static-member-declarations/cmp2/cmp2.component.ts",
  "libs/test-ui-lib/src/lib/static-member-declarations/static-member-declarations.module.ts",
  "libs/test-ui-lib/src/lib/test-button/test-button.component.css",
  "libs/test-ui-lib/src/lib/test-button/test-button.component.html",
  "libs/test-ui-lib/src/lib/test-button/test-button.component.spec.ts",
  "libs/test-ui-lib/src/lib/test-button/test-button.component.stories.ts",
  "libs/test-ui-lib/src/lib/test-button/test-button.component.ts",
  "libs/test-ui-lib/src/lib/test-other/test-other.component.css",
  "libs/test-ui-lib/src/lib/test-other/test-other.component.html",
  "libs/test-ui-lib/src/lib/test-other/test-other.component.spec.ts",
  "libs/test-ui-lib/src/lib/test-other/test-other.component.stories.ts",
  "libs/test-ui-lib/src/lib/test-other/test-other.component.ts",
  "libs/test-ui-lib/src/lib/test-ui-lib.module.ts",
  "libs/test-ui-lib/src/lib/variable-declare/variable-declare-button/variable-declare-button.component.css",
  "libs/test-ui-lib/src/lib/variable-declare/variable-declare-button/variable-declare-button.component.html",
  "libs/test-ui-lib/src/lib/variable-declare/variable-declare-button/variable-declare-button.component.spec.ts",
  "libs/test-ui-lib/src/lib/variable-declare/variable-declare-button/variable-declare-button.component.stories.ts",
  "libs/test-ui-lib/src/lib/variable-declare/variable-declare-button/variable-declare-button.component.ts",
  "libs/test-ui-lib/src/lib/variable-declare/variable-declare-view/variable-declare-view.component.css",
  "libs/test-ui-lib/src/lib/variable-declare/variable-declare-view/variable-declare-view.component.html",
  "libs/test-ui-lib/src/lib/variable-declare/variable-declare-view/variable-declare-view.component.spec.ts",
  "libs/test-ui-lib/src/lib/variable-declare/variable-declare-view/variable-declare-view.component.stories.ts",
  "libs/test-ui-lib/src/lib/variable-declare/variable-declare-view/variable-declare-view.component.ts",
  "libs/test-ui-lib/src/lib/variable-declare/variable-declare.module.ts",
  "libs/test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-anotherview/variable-spread-declare-anotherview.component.css",
  "libs/test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-anotherview/variable-spread-declare-anotherview.component.html",
  "libs/test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-anotherview/variable-spread-declare-anotherview.component.spec.ts",
  "libs/test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-anotherview/variable-spread-declare-anotherview.component.stories.ts",
  "libs/test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-anotherview/variable-spread-declare-anotherview.component.ts",
  "libs/test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-button/variable-spread-declare-button.component.css",
  "libs/test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-button/variable-spread-declare-button.component.html",
  "libs/test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-button/variable-spread-declare-button.component.spec.ts",
  "libs/test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-button/variable-spread-declare-button.component.stories.ts",
  "libs/test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-button/variable-spread-declare-button.component.ts",
  "libs/test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-view/variable-spread-declare-view.component.css",
  "libs/test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-view/variable-spread-declare-view.component.html",
  "libs/test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-view/variable-spread-declare-view.component.spec.ts",
  "libs/test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-view/variable-spread-declare-view.component.stories.ts",
  "libs/test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare-view/variable-spread-declare-view.component.ts",
  "libs/test-ui-lib/src/lib/variable-spread-declare/variable-spread-declare.module.ts",
  "libs/test-ui-lib/src/test-setup.ts",
  "libs/test-ui-lib/tsconfig.json",
  "libs/test-ui-lib/tsconfig.lib.json",
  "libs/test-ui-lib/tsconfig.spec.json",
  "nx.json",
  "package.json",
  "tsconfig.base.json",
  "workspace.json",
]
`;
